tss-esapi/src/structures/pcr/selection.rs

```diff
@@ -44,100 +49,84 @@ impl PcrSelection {
     /// octets that are needed to hold the bit field
     /// that indicate what slots that are selected.
     pub const fn size_of_select(&self) -> PcrSelectSize {
-        self.size_of_select
+        self.pcr_slot_collection.size_of_select()
     }
 
     /// Returns the selected pcrs.
     pub fn selected(&self) -> Vec<PcrSlot> {
-        self.selected_pcrs.iter().collect()
+        self.pcr_slot_collection.collection()
     }
 
     /// Returns true if the specified [PcrSlot] is selected in
     /// the [PcrSelection].
     pub fn is_selected(&self, pcr_slot: PcrSlot) -> bool {
-        self.selected_pcrs.contains(pcr_slot)
+        self.pcr_slot_collection.contains(pcr_slot)
     }
 
     /// Removes the specified [PcrSlot]s from the selected pcrs.
     ///
     /// # Error
     /// If one of the specified pcr slots does not exist in the selected pcrs.
     pub fn deselect_exact(&mut self, pcr_slot: PcrSlot) -> Result<()> {
-        self.remove_exact(pcr_slot.into())
+        self.pcr_slot_collection.remove_exact(pcr_slot)
     }
 
     /// Removes the specified [PcrSlot]s from the selected pcrs.
     pub fn deselect(&mut self, pcr_slot: PcrSlot) {
-        self.remove(pcr_slot.into())
+        self.pcr_slot_collection.remove(pcr_slot)
     }
 
-    /// Merges another [PcrSelection] into this one.
-    pub fn merge(&mut self, other: &Self) -> Result<()> {
+    /// Merges another [PcrSelection] into `self` if the
+    /// elements in the collection does not already exist
+    /// in `self`.
+    ///
+    /// # Constraints
+    /// * Cannot be called with `other` that has a hashing_algorithm
+    ///   that is different from the one in `self`.
+    /// * Cannot be called with `other` that has a size_of_select
+    ///   that is different from the one in `self`.
+    /// * Cannot be called with `other`that contains pcr slots present
+    ///   in `self`.
+    ///
+    /// # Error
```

Comment:
The code uses `# Error` in some places and `# Errors` in others. I think it'd be good to normalize (FTR I'm in favor of the plural even if there is only one error).

Reply:
fixed